collector/adfs.go

```diff
@@ -96,21 +129,245 @@ func newADFSCollector() (Collector, error) {
+			"Total number of successful RP tokens issued over OAuth protocol",
+			nil,
+			nil,
+		),
+		samlPTokenRequests: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "samlp_token_requests_success_total"),
+			"Total number of successful RP tokens issued over SAML-P protocol",
+			nil,
+			nil,
+		),
+		ssoAuthenticationFailures: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "sso_authentications_failure_total"),
+			"Total number of failed SSO authentications",
+			nil,
+			nil,
+		),
+		ssoAuthentications: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "sso_authentications_success_total"),
+			"Total number of successful SSO authentications",
+			nil,
+			nil,
+		),
+		wsfedTokenRequests: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "wsfed_token_requests_success_total"),
+			"Total number of successful RP tokens issued over WS-Fed protocol",
+			nil,
+			nil,
+		),
+		wstrustTokenRequests: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "wstrust_token_requests_success_total"),
+			"Total number of successful RP tokens issued over WS-Trust protocol",
+			nil,
+			nil,
+		),
+		upAuthenticationFailures: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "userpassword_authentications_failure_total"),
+			"Total number of failed AD U/P authentications",
+			nil,
+			nil,
+		),
+		upAuthentications: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "userpassword_authentications_success_total"),
+			"Total number of successful AD U/P authentications",
+			nil,
+			nil,
+		),
+		externalAuthenticationFailures: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "external_authentications_failure_total"),
+			"Total number of failed authentications from external MFA providers",
+			nil,
+			nil,
+		),
+		externalAuthentications: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "external_authentications_success_total"),
+			"Total number of successful authentications from external MFA providers",
+			nil,
+			nil,
+		),
+		artifactDBFailures: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "db_artifact_failure_total"),
+			"Total number of failures connecting to the artifact database",
+			nil,
+			nil,
+		),
+		avgArtifactDBQueryTime: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "db_artifact_query_time_seconds_total"),
+			"Last duration for an artifact database query",
+			nil,
+			nil,
+		),
+		configDBFailures: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "db_config_failure_total"),
+			"Total number of failures connecting to the configuration database",
+			nil,
+			nil,
+		),
+		avgConfigDBQueryTime: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "db_config_query_time_seconds_total"),
+			"Last duration for a configuration database query",
+			nil,
+			nil,
+		),
```

Comment:
Description needs to be updated to state this is the total time of all queries.

Reply:
Updated